Adafruit_BNO055.cpp

```diff
@@ -559,7 +1048,7 @@ bool Adafruit_BNO055::isFullyCalibrated(void)
     @brief  Writes an 8 bit value over I2C
 */
 /**************************************************************************/
-bool Adafruit_BNO055::write8(adafruit_bno055_reg_t reg, byte value)
+int8_t Adafruit_BNO055::write8(adafruit_bno055_reg_t reg, byte value)
```

Comment:
I believe Wire.endTransmission() returns a `byte` which is normally `uint8_t`: https://www.arduino.cc/en/Reference/WireEndTransmission ... see Arduino.h for the typedef.

I'm hesistant to change the return type on this function though since it's already been published. Perhaps we can return 1  or 0 to stick to `bool`logic checking the state of status internally after calling `Wire.endTransmission()`?

Reply:
@dhiltonp thanks for that, I'll use that as a best-practise in future.

@microbuilder I don't have the hardware available to give a 'golden' example, but I could work out something later today. The bit field makes way more sense to me, I didn't have much of a point of reference as to how a user would want those options formatted, that's a much better way of doing things IMO. How best should I make these alterations - Is it bad form to immediately push another master with small changes?

Reply:
There are two possibilities for the updates:

If it's easier for you, you can just take the current master branch here (adafruit/Adafruit_BNO055), fork it again to your github account, make any changes you want, push them to your fork, then create a new PR and I can close this one without merging. It's been so long that maybe you don't have this branch anymore in your local git fork???? This will require you to redo the work in the original PR here though.

OR ... If you still have this code in your own personal fork, just make sure you are up to date with the master branch here, then make your small changes locally, and do a push to your repo and it will be added to this PR with your commit message automatically, and I can merge everything in.

I don't know what is easier for you, but either approach is fine.

Even a very simple example sketch would be good (just place it in the examples folder like the other ones), but I can merge without the example if you don't have time ... it would just be good documentation for other people.